core/src/main/java/io/aiven/klaw/service/RequestService.java

```diff
@@ -0,0 +1,84 @@
+package io.aiven.klaw.service;
+
+import io.aiven.klaw.model.ApiResponse;
+import io.aiven.klaw.model.RequestVerdict;
+import io.aiven.klaw.model.enums.RequestEntityType;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class RequestService {
+
+  @Autowired private SchemaRegstryControllerService schemaRegstryControllerService;
+
+  @Autowired private KafkaConnectControllerService kafkaConnectControllerService;
+
+  @Autowired private TopicControllerService topicControllerService;
+
+  @Autowired private AclControllerService aclControllerService;
+
+  public List<ApiResponse> processApprovalRequests(RequestVerdict requestVerdict) {
+    return requestVerdict.getReqIds().stream()
+        .map(req -> processApprovalRequests(req, requestVerdict.getRequestEntityType()))
+        .collect(Collectors.toList());
+  }
+
+  private ApiResponse processApprovalRequests(String reqId, RequestEntityType requestEntityType) {
+    try {
+      switch (requestEntityType) {
+        case TOPIC:
```

Comment:
```suggestion
        case TOPIC -> topicControllerService.approveTopicRequests(reqId);
```

can be applied on all other switch cases?

Reply:
So i looked at this yesterday and just gave it another look there, the issue is switching to -> won't return the ApiResponse to the calling method only execute the value.

I could add an ApiResponse value and do something like the below but personally think it looks cleaner the other way? Open to your opinion though 

private ApiResponse processApprovalRequests(String reqId, RequestEntityType requestEntityType) {
    try {
      ApiResponse result;
      switch (requestEntityType) {
        case TOPIC -> result = topicControllerService.approveTopicRequests(reqId);
        case ACL -> result = aclControllerService.approveAclRequests(reqId);
        case SCHEMA -> result = schemaRegstryControllerService.execSchemaRequests(reqId);
        case CONNECTOR -> result = kafkaConnectControllerService.approveConnectorRequests(reqId);
        default -> result = ApiResponse.builder()
              .result("Failure Unable to determine target resource.")
              .build();
      }
      return result;
    } catch (Exception ex) {

      return ApiResponse.builder().result("Failure unable to approve requestId " + reqId).build();
    }
  }

Reply:
Using Supplier interface can be an option, but I agree, it's more cleaner the way how you have implemented.